oversweet-api/src/main/java/com/depromeet/oversweet/drink/vo/DrinkStatisticsTotalInfo.java

```diff
@@ -1,23 +1,51 @@
 package com.depromeet.oversweet.drink.vo;
 
 import com.depromeet.oversweet.domain.record.entity.RecordEntity;
-import com.depromeet.oversweet.drink.dto.DrinkDailySugarTotalStatisticsInfo;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
-/*
+/**
     RecordEntity를 이용해서 총합 (당, 칼로리, 남은 당)을 계산하기 위한 클래스입니다.
  */
 public record DrinkStatisticsTotalInfo(
         List<RecordEntity> recordEntities
 ){
-    public DrinkDailySugarTotalStatisticsInfo getTotalStatisticsInfo(final int dailySugar){
+    public DrinkDailySugarTotalStatisticsInfo getDailyTotalStatisticsInfo(final int dailySugar){
         final int totalIntakeSugar = getTotalIntakeSugar();
         final int totalCalorie = getTotalCalorie();
         final int remainingSugar = getRemainingSugar(totalIntakeSugar, dailySugar);
 
         return DrinkDailySugarTotalStatisticsInfo.of(dailySugar, remainingSugar, totalIntakeSugar, totalCalorie);
     }
+
+    public DrinkWeeklySugarTotalStatisticsInfo getWeeklyTotalStatisticsInfo(){
+        final int totalIntakeSugar = getTotalIntakeSugar();
+        final int dailyAverageSugar = getDailyAverageSugar();
+        final int totalCalorie = getTotalCalorie();
+        return new DrinkWeeklySugarTotalStatisticsInfo(totalIntakeSugar, dailyAverageSugar, totalCalorie);
+    }
+
+    public List<DrinkDailySugarInfo> getDayOfWeekSugarInfo() {
+        return recordEntities.stream()
+                .collect(Collectors.groupingBy(record -> record.getCreatedAt().toLocalDate().atStartOfDay()))
+                .entrySet()
+                .stream()
+                .map(entry -> DrinkDailySugarInfo.of(entry.getKey(), getTotalIntakeSugar(entry.getValue())))
+                .toList();
+    }
```

Comment:
Collectors.summingInt 을 조금 더 적극적으로 사용한다면


```suggestion
    public List<DrinkDailySugarInfo> getDayOfWeekSugarInfo() {
        List<DrinkDailySugarInfo> dailySugarInfos = recordEntities.stream()
                .collect(Collectors.groupingBy(
                        record -> record.getCreatedAt().toLocalDate(),
                        Collectors.summingInt(RecordEntity::totalSugar)))
            .entrySet().stream()
            .map(entry -> DrinkDailySugarInfo.of(entry.getKey().atStartOfDay(), entry.getValue()))
            .collect(Collectors.toList());
    }
```

이런식으로 접근할 수 있을 것 같습니다.
위처럼 사용하면 추가 Method인 getTotalIntakeSugar(final List<RecordEntity> entities) 을 사용하지 않아도 될 것 같아요

Reply:
처음에 코드를 작성했을 때는 들어온 날짜를 기준으로 순서를 보장하려고 atStartOfDay()를 사용했던 것 같아요. 수찬님 의견대로 일급컬렉션에서  getTotalIntakeSugar(final List entities)를 사용하는 방법이 더 불필요하다는 생각이 드네요. 감사합니다!